File: Go/demos/log_client/log_client.go
package main

import (
	"fmt"
	"os"
	"os/signal"
	"syscall"

	"api-hub-go/api_hub"
)

func main() {
	// 创建客户端
	client, err := api_hub.NewClient()
	if err != nil {
		panic(err)
	}
	defer client.Close()

	// 准备连接
	client.ResetPrepare()
	if err := client.PrepareClient("ipc:log_service"); err != nil {
		panic(err)
	}
	if ok, _ := client.PrepareDone(); !ok {
		panic("PrepareDone failed")
	}
	defer client.Shutdown()

	// 信号处理（Ctrl+C 优雅退出）
	sigChan := make(chan os.Signal, 1)
	signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
	go func() {
		<-sigChan
		fmt.Println("\nReceived interrupt, shutting down...")
		client.Shutdown()
		os.Exit(0)
	}()

	// 发送第一条日志（INFO）
	h, err := client.CreateDataHnd("log")
	if err != nil {
		panic(err)
	}
	defer client.FreeDataHnd(h)

	if err := client.WriteString(h, "INFO"); err != nil {
		panic(err)
	}
	if err := client.WriteString(h, "Hello from Go client (INFO)"); err != nil {
		panic(err)
	}
	client.Notify("LogService", h)

	// 发送第二条日志（ERROR）
	h2, err := client.CreateDataHnd("log")
	if err != nil {
		panic(err)
	}
	defer client.FreeDataHnd(h2)

	if err := client.WriteString(h2, "ERROR"); err != nil {
		panic(err)
	}
	if err := client.WriteString(h2, "Something went wrong (ERROR)"); err != nil {
		panic(err)
	}
	client.Notify("LogService", h2)

	fmt.Println("Logs sent successfully.")
}

File: rust/examples/log_client.rs
// examples/log_client.rs
use api_hub_rust::*;

fn main() -> Result<()> {
    reset_prepare();
    prepare_client("127.0.0.1:9905", std::ptr::null_mut())?;
    prepare_done()?;

    let mut param = DataHandle::new("log")?;
    param.write_string_null_terminated("INFO")?;
    param.write_string_null_terminated("Hello from Rust log client!")?;
    notify("LogService", param.as_raw());

    let mut param2 = DataHandle::new("log")?;
    param2.write_string_null_terminated("ERROR")?;
    param2.write_string_null_terminated("Something went wrong!")?;
    notify("LogService", param2.as_raw());

    std::thread::sleep(std::time::Duration::from_millis(500));
    exit_main_thread();
    shutdown();
    Ok(())
}